### PR TITLE
180 enrich be alientaxa cube

### LINKED ISSUE #180: enrich be_alientaxa_cube
@SanderDevisscher Which **spatial regions** do you want to show for the invasion maps? 

(1) Municipalities & provinces like in the current *invasion* maps -> need to enrich data file [`be_alientaxa_cube.csv`](https://github.com/inbo/aspbo/blob/uat/data/output/UAT_processing/be_alientaxa_cube.csv) with columns
  - `NAAM`: municipality name
  - `provincie`: province name

(2) UTM 1x1 and 10x10 squares like in the current *observation* maps -> no change to the input data needed

_Originally posted by @mvarewyck in https://github.com/inbo/alien-species-portal/issues/73#issuecomment-2141486593_

### PR DESCRIPTION
**Pull request by soriadelva**
The utm1 and utm10 .gpkg files have been overwritten, appending commune and province information.
There are two things to keep in mind:

- in the UTM files, some pixels are indicated as is.Brussels ==TRUE when there is at least some overlap with Brussels, but note that after assigning communes and provinces based on the highest percentage overlap with the pixel, only few of these (e.g., only 1 in the UTM10 layer) will get assigned brussels as gewest. see 
brussels

- When assigning a gewest based on the percentage overlap with the communes and provinces outline, a few cells will be assigned a different gewest depending on which one is used. This is because we assign them based on the highest percentage overlap, so in some cases this may be a commune in flanders, while at the province level the highest percentage overlap will be with wallonia. Since we only keep one layer of gewest I decided to keep the one based on the left_join with the provinces as this is a higher-level file than the communes. See an example for cells that were not assigned the same gewest: 
not_same_gewest

### CODE REVIEW

File: src/Processing/processing_utm10_files.R
Comment (SanderDevisscher) [inline src/Processing/processing_utm10_files.R:1]: This should be done in the script which updates the be_alientaxa_cube

**Review by SanderDevisscher — CHANGES_REQUESTED** [review]
Based on discussion in https://github.com/inbo/alien-species-portal/issues/73 we should rework the following:

### rework saving the be_alientaxa_cube bypassing github:
- [x] replace https://github.com/inbo/aspbo/blob/09283479dcefec2900cacdddf9032b1f3f441f2b/src/05_occurrence_indicators_preprocessing.Rmd#L61-L62 with the [left join](https://github.com/inbo/aspbo/blob/a9e3a2831ff0af20a587d02be96aa204710091e4/src/Processing/processing_utm10_files.R#L175) & save to AWS bucket logic (see steps below)
- [x] work with temporary directory & file `tempdir()` & `tempfile()`
- [x] cut & paste https://github.com/inbo/aspbo/blob/09283479dcefec2900cacdddf9032b1f3f441f2b/src/upload_processing_to_UAT.R#L83-L84 from [upload_processing_to_UAT.R](https://github.com/inbo/aspbo/blob/uat/src/upload_processing_to_UAT.R) into 5_occurrence_indicators_preprocessing.Rmd
- [x] cut & paste https://github.com/inbo/aspbo/blob/09283479dcefec2900cacdddf9032b1f3f441f2b/src/upload_processing_to_UAT.R#L128-L132 from [upload_processing_to_UAT.R](https://github.com/inbo/aspbo/blob/uat/src/upload_processing_to_UAT.R) into 5_occurrence_indicators_preprocessing.Rmd
- ℹ️ The code above uses the `alienSpecies` package aka the [alien species portal](https://github.com/inbo/alien-species-portal). The yaml already installs this package so no changes needed here !! 
- ℹ️ The `assume role` step, needed to give the action access to the AWS bucket, is already present in the yaml [update_indicators_preprocessing.yaml](https://github.com/inbo/aspbo/blob/uat/.github/workflows/update_indicators_preprocessing.yaml) so no changes needed here !!

Additionally I noted some redundant parts of the code:
### cleanup:
- [x] remove 1 commit & push step https://github.com/inbo/aspbo/blob/09283479dcefec2900cacdddf9032b1f3f441f2b/.github/workflows/update_indicators_preprocessing.yaml#L91-L98 or https://github.com/inbo/aspbo/blob/09283479dcefec2900cacdddf9032b1f3f441f2b/.github/workflows/update_indicators_preprocessing.yaml#L100-L107
This step is still needed to commit the intermediate, datacleaning files.  
- [x] rename scripts to `cube preprocessing` (open for better suggestions)
- [x] rename workflow to `cube preprocessing` (open for better suggestions)
- [x] rename files on the wiki

finally when everything is done we should test the following:
### testing:
- [x] does `cube preprocessing` - workflow succeed when triggered from `180-enrich-be_alientaxa_cube` ? (probably still named `create_timeseries.csv` until merge)
- [x] does `upload_files_processing` - workflow succeed when triggered from `180-enrich-be_alientaxa_cube` ?

⚠️ These changes will interfere with the changes in #199 I would suggest doing a PR from 199 to this branch.

**Comment by SanderDevisscher** [comment]
testing cube preprocessing workflow: [cube preprocessing](https://github.com/inbo/aspbo/actions/workflows/cube_preprocessing.yaml)

testing upload to uat processing: [upload_files_processing](https://github.com/inbo/aspbo/actions/workflows/upload_files_processing.yaml)

**Review by SanderDevisscher — APPROVED** [review]

**Comment by SanderDevisscher** [comment]
@mvarewyck can you confirm the be_alientaxa_cube on the bucket contains the correct fields ?

**Comment by mvarewyck** [comment]
> @mvarewyck can you confirm the be_alientaxa_cube on the bucket contains the correct fields ?

Yes it contains the necessary info, but
* it contains many extra columns which can be dropped to reduce file size: EOFORIGIN, NOFORIGIN, NISCODE_gemeente, NISCODE_provincie, geom 
@soriadelva Are these columns from processing the data and can they be dropped safely?
* is still saved as "be_alientaxa_cube_processed.RData" while version 1.0.0 of `alienSpecies` saves them as .parquet file. No need to change anything here, this will automatically be fine in the future.

**Comment by soriadelva** [comment]
> > @mvarewyck can you confirm the be_alientaxa_cube on the bucket contains the correct fields ?
> 
> Yes it contains the necessary info, but
> 
> * it contains many extra columns which can be dropped to reduce file size: EOFORIGIN, NOFORIGIN, NISCODE_gemeente, NISCODE_provincie, geom
>   @soriadelva Are these columns from processing the data and can they be dropped safely?
> * is still saved as "be_alientaxa_cube_processed.RData" while version 1.0.0 of `alienSpecies` saves them as .parquet file. No need to change anything here, this will automatically be fine in the future.

@mvarewyck, indeed these can be dropped safely. Which columns would you want me to keep from your side?

**Comment by mvarewyck** [comment]
> @mvarewyck, indeed these can be dropped safely. Which columns would you want me to keep from your side?

These are sufficient
```
> colnames(taxData)
 [1] "year"           "cell_code1"     "taxonKey"       "n"             
 [5] "isFlanders"     "isWallonia"     "isBrussels"     "gemeente"      
 [9] "provincie"      "gewest"         "scientificName" "classKey"      
[13] "cell_code10"   
```

**Comment by soriadelva** [comment]
> > @mvarewyck, indeed these can be dropped safely. Which columns would you want me to keep from your side?
> 
> These are sufficient
> 
> ```
> > colnames(taxData)
>  [1] "year"           "cell_code1"     "taxonKey"       "n"             
>  [5] "isFlanders"     "isWallonia"     "isBrussels"     "gemeente"      
>  [9] "provincie"      "gewest"         "scientificName" "classKey"      
> [13] "cell_code10"   
> ```

@mvarewyck some of these are not present in the be_alientaxa_cube file, the colnames present are the following:
```
> colnames(be_alientaxa_cube)
 [1] "year"                  "eea_cell_code"         "taxonKey"              "n"                     "min_coord_uncertainty" "isFlanders"            "isWallonia"           
 [8] "isBrussels"            "EOFORIGIN"             "NOFORIGIN"             "NISCODE_gemeente"      "gemeente"              "NISCODE_provincie"     "provincie"            
[15] "gewest"                "geom"  
```
Should I keep the ones you indicated that are present?

**Comment by mvarewyck** [comment]
> Should I keep the ones you indicated that are present?

No, I think you mention columns from another data set, probably a pre-processed because I recognize `eea_cell_code` while I mentioned the ones from the processed `be_alientaxa_cube_processed.RData` including translated cell codes "cell_code1" and "cell_code10".
So, for your file, you can drop: 

```
"EOFORIGIN"             "NOFORIGIN"             "NISCODE_gemeente"       "NISCODE_provincie"    "geom"  
```

**Comment by soriadelva** [comment]
> > Should I keep the ones you indicated that are present?
> 
> No, I think you mention columns from another data set, probably a pre-processed because I recognize `eea_cell_code` while I mentioned the ones from the processed `be_alientaxa_cube_processed.RData` including translated cell codes "cell_code1" and "cell_code10". So, for your file, you can drop:
> 
> ```
> "EOFORIGIN"             "NOFORIGIN"             "NISCODE_gemeente"       "NISCODE_provincie"    "geom"  
> ```

Yes, indeed. These are the ones from the be_alientaxa_cube file that is placed on the bucket, not the processed file. I'll drop the columns and then this branch should be ready to be merged 👍

**Pull request merged**